File: scala/src/com/runtimeverification/ktest/kthings/stmt.scala
package com.runtimeverification.ktest.kthings

import com.runtimeverification.ktest.KThing
import com.runtimeverification.ktest.kthings.aexp.Aexp

object stmt {
  class Stmt extends KThing
  case class SBlock(sblock: block.Block) extends Stmt
  case class Assign(id: Id, aexp: Aexp) extends Stmt
  case class If(condition: bexp.Bexp, ithen: block.Block, ielse: block.Block) extends Stmt
  case class While(condition: bexp.Bexp, body: block.Block) extends Stmt
  case class Sequence(first: stmt.Stmt, second: stmt.Stmt) extends Stmt
}

File: java.optimized/src/main/java/com/runtimeverification/ktest/nonterminals/Stmt.java
package com.runtimeverification.ktest.nonterminals;

import com.runtimeverification.ktest.KThing;

public class Stmt {
    public static KThing block(KThing block) {
        return new KThing(KThing.Type.STMT_BLOCK, block);
    }

    public static KThing assign(KThing id, KThing aexp) {
        return new KThing(KThing.Type.STMT_ASSIGN, id, aexp);
    }

    public static KThing sequence(KThing first, KThing second) {
        return new KThing(KThing.Type.STMT_SEQUENCE, first, second);
    }

    public static KThing swhile(KThing condition, KThing block) {
        return new KThing(KThing.Type.STMT_WHILE, condition, block);
    }

    public static KThing sif(KThing condition, KThing ithen, KThing ielse) {
        return new KThing(KThing.Type.STMT_IF, condition, ithen, ielse);
    }
}
